citesphere/src/main/java/edu/asu/diging/citesphere/web/user/ShowPeopleController.java

```diff
@@ -0,0 +1,125 @@
+        } catch (NumberFormatException ex) {
+            logger.warn("Trying to access invalid page number: " + page);
+        }
+        IUser user = (IUser) authentication.getPrincipal();
+        CitationResults results;
+        try {
+            results = citationManager.getGroupItems(user, groupId, null, pageInt, sort);
+        } catch (ZoteroHttpStatusException e) {
+            logger.error("Exception occured", e);
+            return "error/500";
+        } catch (GroupDoesNotExistException e) {
+            logger.error("Exception occured", e);
+            return "error/404";
+        }
+        model.addAttribute("zoteroGroupId", groupId);
+        model.addAttribute("currentPage", pageInt);
+        model.addAttribute("sort", sort);
+
+        Map<IPerson, List<ICitation>> personCitationMap = new HashMap<IPerson, List<ICitation>>();
+
+        results.getCitations().stream()
+                .forEach(
+                        iCitation -> (Stream
+                                .of(iCitation.getAuthors().stream(), iCitation.getEditors().stream(),
+                                        iCitation.getOtherCreators().stream()
+                                                .map(iOtherCreator -> iOtherCreator.getPerson()))
+                                .reduce(Stream::concat).orElseGet(Stream::empty))
+                                        .filter(distinctByKey(p -> p.getFirstName())).forEach(au -> {
+                                            boolean containKey = false;
+                                            for (IPerson key : personCitationMap.keySet()) {
+                                                if (key.getFirstName().equals(au.getFirstName())
+                                                        && key.getLastName().equals(au.getLastName())) {
+                                                    personCitationMap.get(key).add(iCitation);
+                                                    containKey = true;
+                                                    break;
+                                                }
+                                            }
+                                            if (!containKey) {
+                                                List<ICitation> citList = new ArrayList<ICitation>();
+                                                citList.add(iCitation);
+                                                personCitationMap.put(au, citList);
+                                            }
+
+                                        }));
+
+        Map<IPerson, List<ICitation>> personCitationTMap = new TreeMap<IPerson, List<ICitation>>(personCitationMap);
+
+        Collection<IPerson> keyCollection = personCitationTMap.keySet();
+        List<IPerson> peopleList = new ArrayList<IPerson>(keyCollection);
+
+        model.addAttribute("people", peopleList);
+
+        personCitationSortedMap = new TreeMap<IPerson, List<ICitation>>(personCitationTMap);
```

Comment:
all of the above is business logic and should not be in the controller

Reply:
also, spring beans are singletons by default. If someone else goes to a person index of another group, this will be overwritten with their map.